> F:\workspace.1\_work\1\s\src\Installers\Windows\Wix.targets(68,5): error MSB3030: Could not copy the file "F:\workspace.1\_work\1\s\artifacts\bin\TargetingPack\x86\Release\AspNetCoreTargetingPack-x86.msi" because it was not found. [F:\workspace.1\_work\1\s\src\Installers\Windows\TargetingPack\TargetingPack.wixproj]

> error : &#x27;&#x2F;home&#x2F;vsts&#x2F;work&#x2F;1&#x2F;s&#x2F;artifacts&#x2F;obj&#x2F;TargetingPack.Layout&#x2F;Release&#x2F;&#x27; directory does not exist [&#x2F;home&#x2F;vsts&#x2F;work&#x2F;1&#x2F;s&#x2F;src&#x2F;Installers&#x2F;Debian&#x2F;TargetingPack&#x2F;Debian.TargetingPack.debproj]‌

And some test failures due to the ref pack not being present:

`PlatformManifestListsAllFiles`
> System.IO.DirectoryNotFoundException : Could not find a part of the path 'F:\\workspace\\_work\\1\\s\\artifacts\\obj\\TargetingPack.Layout\\Release\\packs\\Microsoft.AspNetCore.App.Ref\\3.0.1-ci\\data\\PlatformManifest.txt'.

Plus, all template tests fail because they depend on "restoring" the live-built ref-pack:

> Project new reactredux failed to publish.\r\nProcess exited with code 1\nStdErr: \nStdOut: Microsoft (R) Build Engine version 16.3.0+0f4c62fea for .NET Core\r\nCopyright (C) Microsoft Corporation. All rights reserved.\r\n\r\nF:\\workspace\\_work\\1\\s\\src\\ProjectTemplates\\test\\bin\\Release\\netcoreapp3.0\\TestTemplates\\AspNet.reactredux.okac0s5wov4\\AspNet.reactredux.okac0s5wov4.csproj : error NU1102: Unable to find package Microsoft.AspNetCore.App.Ref with version (= 3.0.1-ci)

Would it make more sense to disable _publishing_ the ref pack, rather than disabling building it?

--- a/Directory.Build.props
+++ b/Directory.Build.props
@@ -91,7 +91,7 @@
     <TargetingPackInstallerBaseName>aspnetcore-targeting-pack</TargetingPackInstallerBaseName>
 
     <!-- Used to only produce targeting pack installers/packages once per major.minor. -->
-    <IsTargetingPackBuilding Condition="'$(AspNetCorePatchVersion)' == '0' AND '$(DotNetBuildFromSource)' != 'true'">false</IsTargetingPackBuilding>
+    <IsTargetingPackBuilding Condition="'$(AspNetCorePatchVersion)' != '0' OR '$(DotNetBuildFromSource)' == 'true'">false</IsTargetingPackBuilding>
 
     <!--
       Archives and installers using this prefix are intended for internal-use only.

--- a/src/Framework/ref/Microsoft.AspNetCore.App.Ref.csproj
+++ b/src/Framework/ref/Microsoft.AspNetCore.App.Ref.csproj
@@ -85,7 +85,7 @@ This package is an internal implementation of the .NET Core SDK and is not meant
     </BuildDependsOn>
 
     <!-- No-op when in source build -->
-    <BuildDependsOn Condition="'$(IsTargetingPackBuilding)' != 'false' and '$(DotNetBuildFromSource)' == 'true'"/>
+    <BuildDependsOn Condition="'$(IsTargetingPackBuilding)' != 'false' or '$(DotNetBuildFromSource)' == 'true'"/>
   </PropertyGroup>
 
   <!-- Override the default MSBuild targets so that nothing is returned from the project since it represents a collection of assemblies. -->

--- a/src/Framework/test/TargetingPackTests.cs
+++ b/src/Framework/test/TargetingPackTests.cs
@@ -56,6 +56,7 @@ namespace Microsoft.AspNetCore
         }
 
         [Fact]
+        [Skip]
         public void PlatformManifestListsAllFiles()
         {
             var platformManifestPath = Path.Combine(_targetingPackRoot, "data", "PlatformManifest.txt");

--- a/src/Installers/Debian/Directory.Build.targets
+++ b/src/Installers/Debian/Directory.Build.targets
@@ -49,7 +49,9 @@
 
     <!-- Build SharedFx Bundle Deb package -->
 
-    <Exec Command="$(DebianBuildScript) -i '$(IntermediateOutputPath)' -o '$(IntermediateOutputPath)out/' -C '$(PackageContentRoot)'" />
+    <Exec 
+      Condition="'$(IsTargetingPackBuilding)' != 'false'"
+      Command="$(DebianBuildScript) -i '$(IntermediateOutputPath)' -o '$(IntermediateOutputPath)out/' -C '$(PackageContentRoot)'" />
 
     <PropertyGroup>
       <BuildScriptOutputFileName>$(PackageId)_$(DebPackageVersion)-$(PackageRevision)_$(DebianPackageArch).deb</BuildScriptOutputFileName>

--- a/src/Installers/Windows/TargetingPack/TargetingPack.wixproj
+++ b/src/Installers/Windows/TargetingPack/TargetingPack.wixproj
@@ -9,6 +9,7 @@
     <OutputName>$(Name)-$(Platform)</OutputName>
     <OutputType>Package</OutputType>
     <IsShipping>true</IsShipping>
+    <SkipCopyToArtifactsDirectory Condition="'$(IsTargetingPackBuilding)' == 'false'">true</SkipCopyToArtifactsDirectory>
     <ProjectGuid>0AC34F1B-8056-4FFB-A398-E6BB7D67B48D</ProjectGuid>
     <HarvestDirectoryAutoGenerateGuids>true</HarvestDirectoryAutoGenerateGuids>
     <HarvestDirectorySuppressSpecificWarnings>5150;5151</HarvestDirectorySuppressSpecificWarnings>

--- a/src/Installers/Windows/Wix.targets
+++ b/src/Installers/Windows/Wix.targets
@@ -63,7 +63,7 @@
   </Target>
 
   <Target Name="CopyToArtifactsDirectory"
-          Condition=" '$(IsShipping)' == 'true' "
+          Condition=" '$(IsShipping)' == 'true' AND '$(SkipCopyToArtifactsDirectory)' != 'true' "
           AfterTargets="Build">
     <Copy SourceFiles="$(TargetPath)" DestinationFiles="$(InstallersOutputPath)$(PackageFileName)" />
     <ItemGroup>

